base.py

```
@@ -498,10 +498,21 @@ def update_time(scene):
 class MBDynReadLog(bpy.types.Operator):
     """ Imports MBDyn nodes and elements by parsing the .log file """
     bl_idname = "animate.read_mbdyn_log_file"
-    bl_label = "MBDyn .log file parsing"
```

why are you changing this label?

---

That is the label which shows up in the dialog box. I couldn't figure out a way to set the title of the dialog box any other way.

---

Ah, I see. Maybe it would be better to move the code for the dialog box in another operator, that gets invoked by the `MBDynRealLog` operator?

---

I spent a couple of hours trying that, and it is turning out to be a mess.
I'm facing a problem as to how to call the invoke method of the dialog box Operator class, from the invoke method of the `MBDynReadLog` class.

---

If I may, I don't see a problem w.r.t the user, as the user reads that label only when the dialog box pops up. I agree it looks confusing to a developer to read, but maybe we could document it properly?

---

I agree in principle but it still seems like a dirty trick to me. 

Let's take a different approach: we can use a single global Enum property, adding it to the collection `MBDynSettingsScene`, for example. The possible values are `DO NOTHING`, `HIDE` and `DELETE`. 

Then we just warn the users (using the standard `report` method) that some nodes were removed from the model. They can change the behaviour for the next time the `.log` file is loaded.

This leads me to another feature that I'd like to add to to Blendyn: collecting all the `report` messages into a  log file (that could be just relative to the current session) that is written in the Text Editor... But I'm going off to a tangent here :)

---

Do keep the feature ideas coming. My proposal is basically writing itself . :)

I don't quite understand your third paragraph. We do the first action without input from the user?

---

Yes, we assume (documenting our choice in the wiki) that the users have already chosen the proper action before reloading the `.log` file, or that they are ok with the default one.

---

I am still not sure about how the user actually does the 'choosing'. Is it still through a dialog box?

---

No I'd rather add the Enum property to the ones shown in the `MBDynImportPanel`. It should have a telling name and description, obviously.

---

I get it now. I didn't really get you the first time.
That seems like a far more simpler solution. I assume the default option according to you is `HIDE`?

---

Yes, i think that it makes more sense that way.